packages/client/lib/service/txpool.ts

```diff
@@ -729,22 +732,45 @@ export class TxPool {
       byNonce.set(address, txs.slice(1))
     }
     // Merge by replacing the best with the next from the same account
+    let blobsCount = 0
     while (byPrice.length > 0) {
       // Retrieve the next best transaction by price
       const best = byPrice.remove()
       if (!best) break
+
       // Push in its place the next transaction from the same account
       const address = best.getSenderAddress().toString().slice(2)
       const accTxs = byNonce.get(address)!
-      if (accTxs.length > 0) {
-        byPrice.insert(accTxs[0])
-        byNonce.set(address, accTxs.slice(1))
+
+      // Insert the best tx into byPrice if
+      //   i) this is not a blob tx,
+      //   ii) or there is no blobs limit provided
+      //   iii) or blobs are still within limit if this best tx's blobs are included
+      //     (it is possible that there is a tx with lets say 10 blobs which can never be included
```

Comment:
may be in a future hardfork when the limit is reached (only for txs which have higher number of blobs than block limit), but yes can be removed as well as I am am fine with that as well

For other txs which have correct limit, but can't be included in this block because some other blobs were included, will be included in the next minted block (hopefully)

Reply:
I am not sure how other mempools work, but I think all txs are invalidated if they are invalid in current circumstances. I don't know if we have other txs which can never be included but are in txpool, but this opens a vector where someone "just" spams us with txs which can never be included and then flooding our mempool until it is full (do we have an expiry mechanism?)

Reply:
hmm we can reject those txs right at the rpc endpoints and let the flexibility remain here